shiny/_docstring.py

```diff
@@ -88,25 +116,40 @@ def _(func: F) -> F:
             ex_dir_found = find_api_examples_dir(func_dir)
 
             if ex_dir_found is None:
-                raise ValueError(
+                raise FileNotFoundError(
                     f"No example directory found for {fn_name} in {func_dir} or its parent directories."
                 )
             example_dir = os.path.join(ex_dir_found, fn_name)
         else:
-            example_dir = os.path.join(func_dir, ex_dir)
+            example_dir = os.path.abspath(os.path.join(func_dir, ex_dir))
 
-        example_file = os.path.join(example_dir, app_file)
-        if not os.path.exists(example_file):
-            raise ValueError(
-                f"No example for {fn_name} found in '{os.path.abspath(example_dir)}'."
+            if not os.path.exists(example_dir):
+                raise FileNotFoundError(
+                    f"Example directory '{example_dir}' does not exist for {fn_name}."
+                )
+
+        app_file_name = app_file or "app.py"
```

Comment:
Not that we need to do anything here, but one "gotcha" with `or` is `"foo" == "" or "foo"`

Reply:
yeah that actually seems desirable in this case